### PR TITLE
update device mesh to include slicing

### PR DESCRIPTION
**Pull request by wz337**
## Description
Update DeviceMesh tutorial to include nD slicing.

### CODE REVIEW

File: recipes_source/distributed_device_mesh.rst [hunk 1]
```diff
@@ -148,6 +148,26 @@ Then, run the following `torch elastic/torchrun <https://pytorch.org/docs/stable
 
     torchrun --nproc_per_node=8 hsdp.py
 
+How to use DeviceMesh for your custom parallel solutions
+--------------------------------------------------------
+When working with large scale training, one can have a more complicated custom parallel training composition. For example, one may need to slice out submeshes for different parallelism solutions.
```
Comment (svekars) [inline recipes_source/distributed_device_mesh.rst:153]:
```suggestion
When working with large scale training, you might have more complex custom parallel training composition. For example, you may need to slice out sub-meshes for different parallelism solutions.
```

**Review by svekars — APPROVED** [review]

File: recipes_source/distributed_device_mesh.rst (hunk 1 above)
Comment (svekars) [reply recipes_source/distributed_device_mesh.rst:153]: If this is for 2.4, can you please label with the 2.4 label.

Comment (wz337) [reply recipes_source/distributed_device_mesh.rst]: @svekars Not sure why but it seems I can't add reviewer and labels. I was gonna add you and one team member @wanchaol as the reviewer.

File: recipes_source/distributed_device_mesh.rst [hunk 2]
```diff
@@ -148,6 +148,26 @@ Then, run the following `torch elastic/torchrun <https://pytorch.org/docs/stable
...
+
+    from torch.distributed.device_mesh import init_device_mesh
+    mesh_3d = init_device_mesh("cuda", (2, 2, 2), mesh_dim_names=("replicate", "shard", "tp"))
+
+    # Users can slice child meshes from the parent mesh.
+    hsdp_mesh = mesh_3d["replicate", "shard"]
+    tp_mesh = mesh_3d["tp"]
+
+    # Users can access the underlying process group thru `get_group` API.
+    replicate_group = hsdp_mesh["Replicate"].get_group()
```
Comment (wanchaol) [inline recipes_source/distributed_device_mesh.rst:166]: these strs need to be lower case

**Pull request merged**